services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js

```diff
@@ -0,0 +1,327 @@
+import React, { Component } from "react";
+import { withRouter } from "react-router-dom";
+import PropTypes from "prop-types";
+import { Button, Menu, Dropdown, Timeline, Result, Row, Col, notification } from "antd";
+import {
+  DownOutlined,
+  ClockCircleOutlined,
+  CheckCircleOutlined,
+  StopOutlined,
+  RightCircleOutlined,
+  LinkOutlined,
+} from "@ant-design/icons";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { getDropdownNoticeOfWorkApplicationStatusCodes } from "@common/selectors/staticContentSelectors";
+import { getNOWProgress } from "@common/selectors/noticeOfWorkSelectors";
+import {
+  updateNoticeOfWorkStatus,
+  fetchApplicationDelay,
+  fetchImportedNoticeOfWorkApplication,
+} from "@common/actionCreators/noticeOfWorkActionCreator";
+import CustomPropTypes from "@/customPropTypes";
+import { modalConfig } from "@/components/modalContent/config";
+import { openModal, closeModal } from "@common/actions/modalActions";
+import NOWStatusIndicator from "@/components/noticeOfWork/NOWStatusIndicator";
+import { getDraftPermitAmendmentForNOW } from "@common/selectors/permitSelectors";
+import { fetchDraftPermitByNOW } from "@common/actionCreators/permitActionCreator";
+import * as route from "@/constants/routes";
+
+/**
+ * @class ProcessPermit - Process the permit. We've got to process this permit. Process this permit, proactively!
+ */
+
+const propTypes = {
+  mineGuid: PropTypes.string.isRequired,
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
+  openModal: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  fetchApplicationDelay: PropTypes.func.isRequired,
+  noticeOfWork: CustomPropTypes.importedNOWApplication.isRequired,
+  updateNoticeOfWorkStatus: PropTypes.func.isRequired,
+  progress: PropTypes.objectOf(PropTypes.any).isRequired,
+  progressStatusCodes: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
+  draftAmendment: CustomPropTypes.permit.isRequired,
+  fetchDraftPermitByNOW: PropTypes.func.isRequired,
+  fetchImportedNoticeOfWorkApplication: PropTypes.func.isRequired,
+};
+
+const defaultProps = {};
```

Comment:
Remove?

Reply:
IMO I'd prefer we have it in every file for consistency and to make life easier when we do need defaultprops, but removed.